### PR TITLE
Fix Elasticsearch GC logs ingestion from Docker logs

### LINKED ISSUE #16583: [Filebeat] Elasticsearch `gc` fileset collects non-GC logs in container environment
Offshoot of #16540. Specifically, see https://github.com/elastic/beats/issues/16540#issuecomment-590988874.

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR fixes the ingestion of Elasticsearch GC logs when Elasticsearch is running in Docker. 

## Why is it important?

Before this PR, the `elasticsearch.gc` fileset was collecting _all_ Elasticsearch logs emitted by the Docker container running Elasticsearch.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works

## Related issues

- Fixes elastic/beats#13164
- Fixes elastic/beats#16583

### CODE REVIEW

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ycombinator** [comment]
jenkins, test this

**Comment by ycombinator** [comment]
jenkins, test this

File: filebeat/module/elasticsearch/gc/test/test-docker.log [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+{"type": "server", "timestamp": "2020-03-20T18:14:43,115Z", "level": "INFO", "component": "o.e.c.m.MetaDataMappingService", "cluster.name": "docker-cluster", "node.name": "fbfaf8ed00d1", "message": "[foo/OZpA00otT8KizSvZvm9N3g] create_mapping", "cluster.uuid": "PDU06UHpSoCo4LBnmGO5ag", "node.id": "Vn8rKzFyR2i8Qpp_l9IMxQ"  }
+2020-03-20T18:14:44,032Z: 14597.826: [GC (CMS Initial Mark) [1 CMS-initial-mark: 131804K(174784K)] 142444K(253440K), 0.0021716 secs] [Times: user=0.01 sys=0.00, real=0.00 secs]
+{"type": "server", "timestamp": "2020-03-20T18:14:54,832Z", "level": "DEBUG", "component": "o.e.a.b.TransportShardBulkAction", "cluster.name": "docker-cluster", "node.name": "fbfaf8ed00d1", "message": "[foo][0] failed to execute bulk item (create) index {[foo][12cn-XABDfxaijeBuNtD], source[{ \"foo\": \"bar }]}", "cluster.uuid": "PDU06UHpSoCo4LBnmGO5ag", "node.id": "Vn8rKzFyR2i8Qpp_l9IMxQ" ,
```
[the remark this answers is not in the document]
Comment (ycombinator) [reply filebeat/module/elasticsearch/gc/test/test-docker.log]: No, these are the actual log lines that are generated if you start Elasticsearch in a container. I just took them as-is and pasted them into this sample file.

[the remark this answers is not in the document]
Comment (ycombinator) [reply filebeat/module/elasticsearch/gc/test/test-docker.log]: If you want to try it yourself, it's pretty easy. Just run:

```
docker run --name elasticsearch -p 9200:9200 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT
```

[the remark this answers is not in the document]
Comment (ycombinator) [reply filebeat/module/elasticsearch/gc/test/test-docker.log]: Sorry, I misunderstood your comment. I see that the `{"type": "server", ...` is part of the _expected_ log file, which is incorrect. Thanks for catching this, will fix!

[the remark this answers is not in the document]
Comment (ycombinator) [reply filebeat/module/elasticsearch/gc/test/test-docker.log]: Good 👁  on this @kaiyan-sheng! Fixed via ca4fb45 and 93e1e0b.

[the remark this answers is not in the document]
Comment (kaiyan-sheng) [reply filebeat/module/elasticsearch/gc/test/test-docker.log]: Thanks! It looks good now!

**Comment by ycombinator** [comment]
Travis CI is green and Jenkins CI failures are unrelated. Merging.

**Pull request merged**